packages/compose/src/hooks/use-resize-observer/index.ts

```diff
@@ -0,0 +1,186 @@
+/**
+ * WordPress dependencies
+ */
+import { useRef, useLayoutEffect } from '@wordpress/element';
+/**
+ * Internal dependencies
+ */
+import useEvent from '../use-event';
+import type { ObservedSize } from './_legacy';
+import _useLegacyResizeObserver from './_legacy';
+/**
+ * External dependencies
+ */
+import type { ReactElement } from 'react';
+
+/** `useResizeObserver` options. */
+export type ObserveElementSizeOptions< T extends HTMLElement > =
+	ResizeObserverOptions & {
+		/**
+		 * The target element to observe. This parameter is an alternative to the
+		 * returned ref. The element can be changed dynamically.
+		 */
+		targetElement?: T | undefined | null;
+	};
```

Comment:
Let's not extend the platform type -- what if a future standard adds `targetElement`? If the hook is using platform types, it should use them 1:1 without any confusing extensions. That also applies to `ResizeObserverCallback`.

Reply:
We already have this debate earlier in the thread. I made some arguments regarding why I'd like to keep it. Can we continue discussing there instead of opening another thread here? Feel free to copy this comment there to keep the conversation going, but please address mine first :)